### PR TITLE
add docs for format: sql unit testing

### LINKED ISSUE #5289: format: sql for unit tests - dbt-core Issue #9405
At a minimum, update body to include a link to the page on docs.getdbt.com requiring updates and what part(s) of the page you would like to see updated.

### PR DESCRIPTION
**Pull request by graciegoheen**
## What are you changing in this pull request and why?

Closes #5289

Coming soon (see core PR [here](https://github.com/dbt-labs/dbt-core/pull/9873)) we are introducing a new format for defining mock data for unit tests - `format: sql`. 

This PR adds documentation for that new format.

## Checklist

- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

### CODE REVIEW

**Review by MichelleArk — APPROVED** [review]

**Comment by graciegoheen** [comment]
Let's wait to merge this till after the 17th when we've cut the 3rd beta of 1.8!

File: website/docs/reference/resource-properties/data-formats.md [hunk 1]
```diff
@@ -57,4 +60,44 @@ unit_tests:
         format: csv
         fixture: my_model_a_fixture
 
-```
\ No newline at end of file
+```
+
+## sql
+
+This format:
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-formats.md]:
```suggestion
Using this format:
```

File: website/docs/reference/resource-properties/data-formats.md [hunk 2]
```diff
@@ -57,4 +60,44 @@ unit_tests:
...
+- provides more flexbility for the types of data you can unit test
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-formats.md]:
```suggestion
- Provides more flexibility for the types of data you can unit test
```

File: website/docs/reference/resource-properties/data-formats.md [hunk 3]
```diff
@@ -57,4 +60,44 @@ unit_tests:
...
+- allows you to unit test a model that depends on an ephemeral model
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-formats.md]:
```suggestion
- Allows you to unit test a model that depends on an ephemeral model
```

File: website/docs/docs/build/unit-tests.md [hunk 4]
```diff
@@ -117,9 +117,9 @@ unit_tests:
 ```
 </file>
 
-The previous example defines the mock data using the inline `dict` format, but you can also use `csv` either inline or in a separate fixture file. 
+The previous example defines the mock data using the inline `dict` format, but you can also use `csv` or `sql` either inline or in a separate fixture file. 
 
-You only have to define the mock data for the columns you care about. This enables you to write succinct and _specific_ unit tests.
+When using the `dict` or `csv` format, you only have to define the mock data for the columns you care about. This enables you to write succinct and _specific_ unit tests.
```
Comment (matthewshaver) [inline website/docs/docs/build/unit-tests.md]:
```suggestion
When using the `dict` or `csv` format, you only have to define the mock data for the columns relevant to you. This enables you to write succinct and _specific_ unit tests.
```

File: website/docs/reference/resource-properties/data-formats.md [hunk 5]
```diff
@@ -57,4 +60,44 @@ unit_tests:
...
+  - name: test_my_model
+    model: my_model
+    given:
+      - input: ref('my_model_a')
+        format: sql
+        fixture: my_model_a_fixture
+
+```
+
+Note: Jinja is not supported in SQL fixtures for unit tests.
```
Comment (matthewshaver) [inline website/docs/reference/resource-properties/data-formats.md]:
```suggestion
**Note:** Jinja is unsupported in SQL fixtures for unit tests.
```

**Review by matthewshaver — APPROVED** [review]

**Pull request merged**